core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/speech/recognizer/GoogleSpeechRecognizer.java

```diff
@@ -0,0 +1,128 @@
+ * limitations under the License.
+ *
+ * For more information see notice.md
+ */
+
+package de.unistuttgart.iaas.amyassist.amy.core.speech.recognizer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.unistuttgart.iaas.amyassist.amy.core.speech.resulthandler.AbstractSpeechResultHandler;
+import de.unistuttgart.iaas.amyassist.amy.remotesr.RemoteSR;
+import de.unistuttgart.iaas.amyassist.amy.remotesr.RemoteSRListener;
+
+/**
+ * Calls the remote SR Service
+ * 
+ * @author Kai Menzel
+ */
+public class GoogleSpeechRecognizer implements RemoteSRListener, SpeechRecognizerType {
+
+	// --------------------------------------------------------------
+	// Dependencies
+
+	// --------------------------------------------------------------
+	// Fields
+
+	private static final int WAITING_FOR_CHROME_TO_START_TIME = 5000;
+
+	/**
+	 * logger for all the Speech Recognition classes
+	 */
+	private final Logger logger = LoggerFactory.getLogger(GoogleSpeechRecognizer.class);
+
+	/**
+	 * Handler who use the translated String for commanding the System
+	 */
+	private AbstractSpeechResultHandler resultHandler;
+
+	/**
+	 * The Google Speech Service
+	 */
+	private RemoteSR recognizer;
+
+	// --------------------------------------------------------------
+	// Construcor
+
+	/**
+	 * Returns Recognition Result to given Handler from Given Recognizer
+	 * 
+	 * @param recognizer
+	 *            The Remote Recognizer
+	 */
+	public GoogleSpeechRecognizer(RemoteSR recognizer) {
+		this.recognizer = recognizer;
+
+		this.recognizer.setListener(this);
+	}
+
+	// --------------------------------------------------------------
+	// Init
+
+	// --------------------------------------------------------------
+	// Methods
+
+	/**
+	 * @see de.unistuttgart.iaas.amyassist.amy.core.speech.recognizer.SpeechRecognizerType#getRecognition(de.unistuttgart.iaas.amyassist.amy.core.speech.resulthandler.AbstractSpeechResultHandler)
+	 */
+	@Override
+	public void getRecognition(AbstractSpeechResultHandler usedResultHandler) {
+		this.resultHandler = usedResultHandler;
+
+		if (!this.recognizer.requestSR()) {
+
+			this.logger.info("A problem connecting to the remote Speech Recognition has occured.");
+			while (!this.recognizer.requestSR()) {
+				this.logger.info("trying to reconnect");
+				try {
+					Thread.sleep(WAITING_FOR_CHROME_TO_START_TIME);
+				} catch (InterruptedException e) {
+					this.logger.error("Error with Chrome:", e);
+					Thread.currentThread().interrupt();
+				}
+			}
+			this.logger.info("Reconnected");
+		}
+
+		this.logger.info("waiting for speech input");
+	}
+
+	/**
+	 * @see de.unistuttgart.iaas.amyassist.amy.remotesr.RemoteSRListener#remoteSRDidRecognizeSpeech(java.lang.String)
+	 */
+	@Override
+	public void remoteSRDidRecognizeSpeech(String message) {
+		if (message.isEmpty()) {
+			this.logger.warn("I could not understand that. Can you please repeat your Command");
```

Comment:
This should not be done by each SR Type itself, but better by the result handler.